src/cli.js

```diff
@@ -65,10 +73,31 @@ async function loadPlugin(name) {
   return plugins;
 }
 
+function findPkgJsonConfig() {
+  // fallback to original config name 'wsk' in case anything goes wrong
+  // for safe backwards compatibility
+  let cfg = 'wsk';
+
+  try {
+    // find package.json in parent directories (like yargs does)
+    const pkg = escalade(process.cwd(), (_, names) => (names.includes('package.json') ? 'package.json' : undefined));
+
+    if (pkg) {
+      const pkgJson = JSON.parse(readFileSync(pkg, 'utf8'));
```

Comment:
can we make this async?

Reply:
Yeah but then you have to make `CLI.buildArgs()` also async and maybe more downstream... and I wanted to keep the change limited. IMO it's not bad for a cli which only has "one thread" here anyway?

Reply:
Not arguing either way, but `yargs` itself is very sync-y internally: https://github.com/yargs/yargs/blob/0c95f9c79e1810cf9c8964fbf7d139009412f7e7/lib/yargs-factory.ts#L1637